GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -572,6 +560,23 @@ public void LaunchPostFetchJob(List<string> packIndexes)
             }
         }
 
+        private void InvalidateState(bool invalidateProjection, bool invalidateModifiedPaths)
+        {
+            if (invalidateProjection)
+            {
+                this.GitIndexProjection.InvalidateProjection();
+            }
+
+            if (invalidateModifiedPaths)

```

Comment:
I guess there's actually a bigger question, can `invalidateModifiedPaths` ever be true during a git command that will update the projection?

I don't think that any of this is necessarily an issue, the questions are for me to better understand what's happening :smile:

Reply:
Right now that should never happen since we are passing the UpdatedSkipworktreeBits only when in the reset mixed and update-index paths for git, so something in git would need to change for the UpdatedWorkingDirectory flag to get passed as well because it would need to run through the unpack_trees code which is updating the working directory with index entries and both those commands should never have to do that.